src/Aspire.Hosting/ApplicationModel/ReferenceEnvironmentInjectionFlags.cs

```diff
@@ -25,7 +25,17 @@ public enum ReferenceEnvironmentInjectionFlags
     ConnectionProperties = 1 << 1,
 
     /// <summary>
-    /// Both connection string and connection properties will be injected as environment variables.
+    /// Each endpoint defined on the resource will be injected using the format "services__{resourceName}__{endpointName}__{endpointIndex}".
```

Comment:
> endpointIndex

What is the reason we have an index at the end? How are those scenarios going to work with the `Endpoints` format?